test/SwiftSyntax/AbsolutePosition.swift

```diff
@@ -0,0 +1,63 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+// REQUIRES: OS=macosx
+// REQUIRES: objc_interop
+
+import StdlibUnittest
+import Foundation
+import SwiftSyntax
+import SwiftLang
+
+func getInput(_ file: String) -> URL {
+  var result = URL(fileURLWithPath: #file)
+  result.deleteLastPathComponent()
+  result.appendPathComponent("Inputs")
+  result.appendPathComponent(file)
+  return result
+}
+
+
+class FuncRenamer: SyntaxRewriter {
+  override func visit(_ node: FunctionDeclSyntax) ->DeclSyntax {
+    return (super.visit(node) as! FunctionDeclSyntax).withIdentifier(
+      SyntaxFactory.makeIdentifier("anotherName"))
+  }
+}
+
+var PositionTests = TestSuite("AbsolutePositionTests")
+
+PositionTests.test("Visitor") {
+  expectDoesNotThrow({
+    let content = try SwiftLang.parse(getInput("visitor.swift"))
+    let source = try String(contentsOf: getInput("visitor.swift"))
+    let parsed = try SourceFileSyntax.decodeSourceFileSyntax(content)
+    // By testing this, we assume that the eof token doesn't have leading
+    // trivia
```

Comment:
You're right. I'm also pondering what is the right behavior here. C++ side behavior is tailored for diagnostics. In Swift syntax, I'm leaning towards including trivia to the start position. It feels wrong to point inside a `TokenSyntax`. What do you think? @harlanhaskins

Reply:
Yeah I think it would be very unfortunate for diagnostics to point to trivia instead of the content of the node. It would probably be a good idea to point after the `leadingTrivia`?

Reply:
Ok, I'll change it to point after `leadingTrivia`.